### DISCUSSION TITLE
[getInitialProps with Redux Saga]: How can I await for data being fetched in getInitialProps when dispatching an async action while using redux-saga?

### DISCUSSION THREAD

**Discussion by AntwanSherif** [vercel/next.js#10754] [Help]
Hello,

I'm not sure if there's a similar question here, but I've searched a lot but couldn't find an answer.

I need to know how can I wait for data fetching inside getInitialProps when dispatching an action with redux-saga in use.

All the answers I find make sense when using redux-thunk or do the actual API, but not for redux-saga

**Comment by bekapod** [comment]
I've used https://www.npmjs.com/package/redux-wait-for-action with some success. Dispatch data fetch action then `await` store.dispatch of 'wait for action' before returning from `getInitialProps`.
